### DISCUSSION TITLE
Craft lets you install php incompatible plugins in the control panel

### DISCUSSION THREAD

**Discussion by kaspar-allenbach** [craftcms/cms#8620] [Ideas]
I think the Craft update function should account for the php versions of the plugins and only display plugin updates if it is compatible with the current php version on the server.

Example:

* I have [craft-async-queue](https://github.com/ostark/craft-async-queue) installed.
* Now there is an update for this plugin (which requires unknowingly `php 7.1`).
* my server runs `php 7.0`
* So I hit "update" in the control panel
* Now the entire site breaks because I've installed an incompatible plugin.

So it would be very neat to have a php check for plugins. Or at least a big warning or something visual in the update tab.

### Steps to reproduce

1. install a incompatible plugin
2. read the logs

### Additional info

- Craft version: 3.1.5 (Pro)
- PHP version: 7.0.32
- Database driver & version: MySQL 5.7.23
- Image driver & version: GD 7.0.32
- Plugins & versions: 
    - AsyncQueue 1.4
    - CP Clear Cache 1.0.3
    - Enupal Translate 1.2.0
    - Redactor 2.3.0

**Comment by kaspar-allenbach** [comment]
After more examination it is my fault. I updated everything and then updated my php version. Which then lead to the breaking of the site because now I have an incompatible plugin installed.

Still tough: Some visual warning about the compatibility on the update panel would be nice to have.

**Comment by brandonkelly** [comment]
Yeah I don’t think that you should be able to install a plugin within the Control Panel if the active PHP version isn’t compatible with the plugin – even if Composer won’t complain about it.
